src/core/trie.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {type AddressModule} from "./address";
+import {
+  NodeAddress,
+  type NodeAddressT,
+  EdgeAddress,
+  type EdgeAddressT,
+} from "./graph";
+import * as NullUtil from "../util/null";
+
+type Entry<T> = {map: RecursiveMap<T>, value: ?T};
```

Comment:
It seems that the implementation is unsound when `T` is nullable. You
could avoid this by either requiring that the values be non-null (via
`$NonMaybeType`), or by using a discriminated union, or by hacking
together a discriminated union with a fixed private `Symbol`.

Reply:
used a symbol.